### PR TITLE
2510 uac by sms

### PR DESCRIPTION
**Pull request by ryangrundy7**
# Motivation and Context

* [x] [Context Index](/CODE_GUIDE.md#context-index) has been kept up to date
Adding new tests for the notify service sms fulfilments
# What has changed

- Adding new tests for the notify service sms fulfilments

# How to test?

- Run with other PRs on ticket
# Links

[Trello](Adding new tests for the notify service sms fulfilments)
# Screenshots (if appropriate):

### CODE REVIEW

**Comment by ryangrundy7** [comment]
[Prototype: Send a UAC by SMS by API (13)](https://trello.com/c/uQcwIQcQ/2510-prototype-send-a-uac-by-sms-by-api-13)

File: CODE_GUIDE.md [hunk 1]
```diff
@@ -47,6 +47,8 @@ Every context attribute used by the tests should be described here.
 | pack_code                 | Stores the pack code used for fulfilments or action rules                                        |
 | template                  | Stores the column template used for fulfilments or action rules                                  |
 | telephone_capture_request | Stores the UAC and QID returned by a telephone capture API call                                 |
+| notify_id                 | Stores the ID of the sms template used for the notify service                                 |
```
Comment (NickGrantONS) [inline CODE_GUIDE.md:50]:
```suggestion
| notify_template_id                 | Stores the ID of the sms template used for the notify service                                 |
```

File: acceptance_tests/features/steps/print_file.py [hunk 2]
```diff
@@ -102,3 +103,22 @@ def create_print_template(context, template):
...
+def create_sms_template(context, template):
+    context.template = template
+
+    # By using a unique random pack_code we have better filter options
+    # We can change/remove this if we get UACS differently or a better solution is found
+    context.pack_code = 'pack_code_' + ''.join(random.choices(string.ascii_uppercase + string.digits, k=10))
+    context.notify_id = str(uuid.uuid4())
+    url = f'{Config.SUPPORT_TOOL_API}/smsTemplates'
+    body = {
+        'notifyId': context.notify_id,
```
Comment (NickGrantONS) [inline acceptance_tests/features/steps/print_file.py:118]:
```suggestion
        'notifyTemplateId': context.notify_template_id,
```

File: acceptance_tests/features/steps/print_file.py [hunk 3]
```diff
@@ -102,3 +103,22 @@ def create_print_template(context, template):
...
+
+
+@step('a sms template has been created with template "{template}"')
+def create_sms_template(context, template):
+    context.template = template
+
+    # By using a unique random pack_code we have better filter options
+    # We can change/remove this if we get UACS differently or a better solution is found
+    context.pack_code = 'pack_code_' + ''.join(random.choices(string.ascii_uppercase + string.digits, k=10))
+    context.notify_id = str(uuid.uuid4())
```
Comment (NickGrantONS) [inline acceptance_tests/features/steps/print_file.py:115]:
```suggestion
    context.notify_template_id = str(uuid.uuid4())
```

File: acceptance_tests/features/steps/sms_fulfilment.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                "packCode": context.pack_code,
+            }
+        }
+    }
+
+    response = requests.post(url, json=body)
+    response.raise_for_status()
+
+
+def _check_notify_api_called_with_correct_notify_id(phone_number, notify_id):
```
Comment (NickGrantONS) [inline acceptance_tests/features/steps/sms_fulfilment.py:49]:
```suggestion
def _check_notify_api_called_with_correct_notify_template_id(phone_number, notify_id):
```

File: acceptance_tests/features/steps/sms_fulfilment.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    response = requests.get(f'{Config.NOTIFY_STUB_SERVICE}/log')
+    test_helper.assertEqual(response.status_code, 200, "Unexpected status code")
+    response_json = response.json()
+    test_helper.assertEqual(len(response_json), 1, "Incorrect number of responses")
+    test_helper.assertEqual(response_json[0]["phone_number"], phone_number, "Incorrect phone number")
+    test_helper.assertEqual(response_json[0]["template_id"], notify_id, "Incorrect template Id")
```
Comment (NickGrantONS) [inline acceptance_tests/features/steps/sms_fulfilment.py:55]:
```suggestion
    test_helper.assertEqual(response_json[0]["template_id"], notify_template_id, "Incorrect Gov Notify template Id")
```

File: acceptance_tests/features/steps/sms_fulfilment.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+@step("notify api was called with SMS template")
+def check_notify_api_call(context):
+    _check_notify_api_called_with_correct_notify_id(context.phone_number, context.notify_id)
```
Comment (NickGrantONS) [inline acceptance_tests/features/steps/sms_fulfilment.py:60]:
```suggestion
    _check_notify_api_called_with_correct_notify_template_id(context.phone_number, context.notify_template_id)
```

**Review by NickGrantONS — CHANGES_REQUESTED** [review]
See comments

**Review by NickGrantONS — APPROVED** [review]
ATs passing. Great work

**Review by liamtoozer — APPROVED** [review]
LGTM 👍  Tested locally

**Pull request merged**